src/core/book-keeper.js

```diff
@@ -1,9 +1,277 @@
 'use strict';
 
-class BookKeeper {
-    constructor () {
+var Util = require('./util');
 
+var moment = require('moment');
+
+var metricsEventEmitter = require('../src/core/event/event-bus.js').EventEmitter;
+
+
+var init = function(EventI) {
+
+    EventI.on('DELETE_OLD_SCHEDULE', function(col, days, cb) {
+
+          var query = {};
+          query[toFetch.date] = {
+              $lt: moment().subtract(days, 'days').toISOString()
+          };
+
+        EventI.emit('DATABASE_REMOVE', {
+            type: col,
+            params: query
+        }, function(err, numRemoved) {
+            if (err) {
+                cb(err);
+            } else {
+                cb(numRemoved);
+            }
+        });
+    });
+
+    EventI.on('CLEAN_SHOPPING_RECORDS', function(days, cb) {
+
+              EventI.emit('DELETE_OLD_SCHEDULE',
```

Comment:
Don't call the native emit() function, use emitEvent() which is defined in event-bus.js